bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -79,17 +71,30 @@
     private boolean showTitle;
     private MagentoGraphqlClient magentoGraphqlClient;
 
+    private int navPageCursor = 1;
+    private int navPageSize = 6;
+    private int[] navPages;
+
     @PostConstruct
     private void initModel() {
         // read properties
         showTitle = properties.get(PN_SHOW_TITLE, currentStyle.get(PN_SHOW_TITLE, SHOW_TITLE_DEFAULT));
+        navPageSize = properties.get(PN_PAGE_SIZE, currentStyle.get(PN_PAGE_SIZE, PAGE_SIZE_DEFAULT));
 
         // get product template page
         productPage = Utils.getProductPage(currentPage);
         if (productPage == null) {
             productPage = currentPage;
         }
 
+        //check if pageCursor available in queryString, set to 1 if not.
+        if (request.getParameter("page") == null) {
+            this.navPageCursor = 1;
+        } else {
+            this.navPageCursor = Integer.parseInt(request.getParameter("page"));
```

Comment:
I would add a basic sanity check here. For example, the current page shouldn't be negative.

Reply:
applied check #97182ad